lib/src/okta/okta.module.ts

```diff
@@ -36,12 +32,17 @@ export function oktaAuthFactory(config: OktaConfig): OktaAuth {
     OktaHasAnyGroupDirective,
   ],
   providers: [
-    OktaAuthGuard,
+    OktaAuthConfigService,
     OktaAuthStateService,
+    OktaAuthGuard,
     {
       provide: OKTA_AUTH,
-      useFactory: oktaAuthFactory,
-      deps: [ OKTA_CONFIG ]
+      useFactory: OktaAuthFactoryService.createOktaAuth,
+      deps: [
+        OktaAuthConfigService,
+        [new Optional(), Router],
```

Comment:
Is including the `Optional` instance required? (question applies to both Router and Location)